docs(getting-started): add views page (#7556)
nuxt/framework: docs/content/1.getting-started/3.views.md

# Views

Nuxt provides several component layers to implement the user interface of your application.

## `app.vue`

![The `app.vue` file is the entry point of your application](/img/getting-started/views/app.svg)

> By default, Nuxt will treat this file as the **entrypoint** and render it's content for every route of the application.

Reviewer: ```suggestion
By default, Nuxt will treat this file as the **entrypoint** and render its content for every route of the application.
```

```vue [app.vue]
<template>
  <div>
   <h1>Welcome to the homepage</h1>
  </div>
</template>
```

::alert

> If you are familiar with Vue, you might wonder where is the `main.js` file that creates a Vue app. Nuxt does this behind the scene.

Reviewer: ```suggestion
If you are familiar with Vue, you might wonder where `main.js` is (the file is that normally creates a Vue app). Nuxt does this behind the scene.
```

::

## Components

![Components are reusable pieces of UI](/img/getting-started/views/components.svg)

> Most of the components are reusable pieces of the user interface, like buttons and menus. In Nuxt, you can create these components in the `components/` directory for them to be automatically available across your application without having to explicitly import them.

Reviewer: ```suggestion
Most components are reusable pieces of the user interface, like buttons and menus. In Nuxt, you can create these components in the `components/` directory, and they will be automatically available across your application without having to explicitly import them.
```

::code-group

```vue [App.vue]
<template>
  <div>
    <h1>Welcome to the homepage</h1>
    <AppAlert>
      This is an auto-imported component.
    </AppAlert>
  </div>
</template>
```

```vue [components/AppAlert.vue]
<template>
  <span>
    <slot />
  </span>
</template>
```

::

## Pages

![Pages are views tied to a specific route](/img/getting-started/views/pages.svg)

> Pages represent views use for each specific route pattern. Every file in the `pages/` directory represents a different route displaying it's content.

Reviewer: ```suggestion
Pages represent views use for each specific route pattern. Every file in the `pages/` directory represents a different route displaying its content.
```

To use pages, create `pages/index.vue` file and add `<NuxtPage />` component to the `app.vue` (or remove `app.vue` for default entry). You can now create more pages and their corresponding routes by adding new files in the `pages/` directory.

::code-group

```vue [pages/index.vue]
<template>
  <div>
   <h1>Welcome to the homepage</h1>
    <AppAlert>
      This is an auto-imported component
    </AppAlert>
  </div>
</template>
```

```vue [pages/about.vue]
<template>
  <section>
    <p>This page will be displayed at the /about route.</p>
  </section>
</template>
```

::

::alert
You will learn more about pages in the [Routing section](/getting-started/routing)
::

## Layouts

![Layouts are wrapper around pages](/img/getting-started/views/layouts.svg)

> Layouts are wrapper around pages that contain a common User Interface for several pages, such as a header and footer display. Layouts are Vue files using `<slot />` components to display the **pages** content. The `layout/default.vue` file will be used by default. Custom layouts can be set as part of your page metadata.

Reviewer: ```suggestion
Layouts are wrappers around pages that contain a common User Interface for several pages, such as a header and footer display. Layouts are Vue files using `<slot />` components to display the **page** content. The `layout/default.vue` file will be used by default. Custom layouts can be set as part of your page metadata.
```
